test/base/src/androidTest/kotlin/io/realm/kotlin/test/shared/RealmSetTests.kt

```diff
@@ -0,0 +1,1137 @@
+ * limitations under the License.
+ */
+
+package io.realm.kotlin.test.shared
+
+import io.realm.kotlin.MutableRealm
+import io.realm.kotlin.Realm
+import io.realm.kotlin.RealmConfiguration
+import io.realm.kotlin.entities.Sample
+import io.realm.kotlin.entities.SampleWithPrimaryKey
+import io.realm.kotlin.ext.query
+import io.realm.kotlin.ext.realmSetOf
+import io.realm.kotlin.ext.toRealmSet
+import io.realm.kotlin.query.RealmResults
+import io.realm.kotlin.query.find
+import io.realm.kotlin.test.platform.PlatformUtils
+import io.realm.kotlin.test.util.TypeDescriptor
+import io.realm.kotlin.types.ObjectId
+import io.realm.kotlin.types.RealmInstant
+import io.realm.kotlin.types.RealmObject
+import io.realm.kotlin.types.RealmSet
+import kotlinx.coroutines.async
+import kotlinx.coroutines.flow.takeWhile
+import kotlinx.coroutines.runBlocking
+import kotlin.reflect.KClassifier
+import kotlin.reflect.KMutableProperty1
+import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertContentEquals
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+import kotlin.test.assertFalse
+import kotlin.test.assertNotNull
+import kotlin.test.assertTrue
+import kotlin.test.fail
+
+val setTestSchema = setOf(RealmSetContainer::class)
+
+class RealmSetTests {
+
+    private val descriptors = TypeDescriptor.allSetFieldTypes
+
+    private lateinit var tmpDir: String
+    private lateinit var realm: Realm
+
+    private val managedTesters: List<SetApiTester<*, RealmSetContainer>> by lazy {
+        // descriptors.mapNotNull { // TODO this is just to avoid having to deal with all supported types when testing something very specific
+        descriptors.map {
+            val elementType = it.elementType
+            when (val classifier = elementType.classifier) {
+                RealmObject::class -> RealmObjectSetTester(
+                    realm,
+                    getTypeSafety(
+                        classifier,
+                        false
+                    ) as SetTypeSafetyManager<RealmSetContainer>
+                )
+                ByteArray::class -> ByteArraySetTester(
+                    realm,
+                    getTypeSafety(
+                        classifier,
+                        elementType.nullable
+                    ) as SetTypeSafetyManager<ByteArray>,
+                    classifier
+                )
+                else -> GenericSetTester(
+                    realm,
+                    getTypeSafety(classifier, elementType.nullable),
+                    classifier
+                )
+                // else -> null // TODO this is just to avoid having to deal with all supported types when testing something very specific
+            }
+        }
+    }
+
+    @BeforeTest
+    fun setup() {
+        tmpDir = PlatformUtils.createTempDir()
+        val configuration = RealmConfiguration.Builder(
+            setTestSchema + setOf(
```

Comment:
Seems a bit funny to have a single element set holding the schema for the test .... and then adding a lot of other classes to it to setup the test 🤪

Reply:
SetTestSchema is only used in `getCloseableRealm` and here, couldn't it be removed? That way you remove the union of this two sets here, and you have an explicit schema set in `getCloseableRealm`

Reply:
I'll get rid of it. Perhaps worth thinking about aligning this too when reworking list and set tests again because this approach is also used for list tests.